lib/rspec/mocks/space.rb

```diff
@@ -56,6 +56,8 @@ def raise_lifecycle_message
     # @private
     class Space
       attr_reader :proxies, :any_instance_recorders, :proxy_mutex, :any_instance_mutex
+      attr_accessor :temporarily_suppressing_verification
+      alias :temporarily_suppressing_verification? :temporarily_suppressing_verification
```

Comment:
I'm not sure `Space` is the right place for this state.  Consider: spaces can be nested.  As currently implemented, this would fail:

``` ruby
without_verifying_partial_doubles do
  RSpec::Mocks.with_temporary_scope do
    dbl = instance_double("MyClass")
    allow(dbl).to receive(:some_method_that_would_fail_verification)
  end
end
```

`RSpec::Mocks.with_temporary_scope` creates a nested temporary space, which will have `temporarily_suppressing_verification` set to false.  Unless you specifically want `temporarily_suppressing_verification` not to inherit to nested scopes (in which case, I'd ask for a use case and for this to be well documented) I think it's better to attach this state to `RSpec::Mocks.configuration` where it would be truly global and not have this kind of gotcha.